implementations/rust/ockam/ockam_api/src/lib.rs

```diff
@@ -503,6 +506,30 @@ impl<'a> Deref for CowStr<'a> {
     }
 }
 
+impl<'a> Display for CowStr<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl<'a> PartialEq<&str> for CowStr<'a> {
+    fn eq(&self, other: &&str) -> bool {
+        self.0 == *other
+    }
+}
+
+impl<'a> PartialEq<str> for CowStr<'a> {
+    fn eq(&self, other: &str) -> bool {
+        self.0 == *other
+    }
+}
+
+impl<'a> PartialEq<String> for CowStr<'a> {
+    fn eq(&self, other: &String) -> bool {
+        self.0 == *other
+    }
+}
```

Comment:
Any idea on how to write this with a single `impl`? Or maybe a couple (one for owned and another for borrowed strings)?